src/core/ext/xds/xds_client.cc

```diff
@@ -966,6 +1017,38 @@ XdsClient::XdsChannel::AdsCall::AdsCall(
   streaming_call_->StartRecvMessage();
 }
 
+// Make this channel active iff:
+// 1. Channel is on the list of authority channels
+// 2. Channel is not the last channel on the list (i.e. not the active channel)
+void XdsClient::XdsChannel::AdsCall::FallForwardIfNeededLocked() {
+  for (auto& a : xds_client()->authority_state_map_) {
+    auto channels = a.second.xds_channels;
+    // Skip if channel is active.
+    if (!channels.empty() && channels.back() == xds_channel()) {
+      continue;
+    }
+    auto channel_it =
+        std::find(channels.begin(), channels.end(), xds_channel());
+    // Skip if channel is not on the list
+    if (channel_it == channels.end()) {
+      continue;
+    }
+    gpr_log(GPR_INFO, "[xds_client %p] Falling forward to %s", this,
+            xds_channel()->server_.server_uri().c_str());
+    // Lower priority channels are no longer needed, connection is back!
+    channels.erase(channel_it + 1, channels.end());
+    // Restore all resource subscriptions
```

Comment:
This should not be needed here.  As per comments elsewhere, we should just do this when the watchers are added, even if the channel status is not ok.